kernel/console/kprintf.c

```diff
@@ -89,7 +88,7 @@ uint64_t kprintf_proc_format_string(const char *s, uint64_t *chars_written, __bu
 	// 64 binary digits (for an unsigned 64-bit type) or 63 and a +/- sign (if signed)
 	// possible "b" sigil at end
 	// NULL terminator
-	char output[66];
+	char output[KPRINTF_LEN];
```

Comment:
This value is used only to hold the result of a <something-other-than-a-string> conversion.  Strings referenced in the format string are just copied directly, and anything else will have a string representation of no more than 66 bytes.

I see the reason to have a 1024-byte buffer in the outermost `kprintf()` function which handles actually sending the string to the console, since console writes can be slow so this can save some calls there, but I'm not sure there's any gain to using a larger buffer in the conversion-specifier handling code.

Reply:
Changes reverted